optuna/integration/chainermn.py

```diff
@@ -41,14 +40,14 @@ class _ChainerMNObjectiveFunc(object):
             index.html#communicators>`_.
     """
 
-    def __init__(self, func, comm):
-        # type: (Callable[[ChainerMNTrial, CommunicatorBase], float], CommunicatorBase) -> None
+    def __init__(
+        self, func: "Callable[[ChainerMNTrial, CommunicatorBase], float]", comm: "CommunicatorBase"
```

Comment:
`Callable` and `float` can be accessible in both type check and run time. `ChainerMNTrial` defines later of the code, and I think we use a string here.

```suggestion
        self, func: Callable[["ChainerMNTrial", "CommunicatorBase"], float], comm: "CommunicatorBase"
```